src/FieldWarning/Assets/Units/Component/Weapon/Cannon.cs

```diff
@@ -30,19 +31,21 @@ public class Cannon : IWeapon
         private ParticleSystem _shotEffect;
         private AudioClip _shotSound;
         private float _shotVolume;
+        private System.Random _random;
```

Comment:
This needs to be static or otherwise shared. Otherwise, if we create multiple cannons at the same time, they will share the same random sequence.